### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Kevalkukadiya**

### CODE REVIEW

File: rest_api/api/auth.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from rest_framework.authentication import BasicAuthentication
+from django.contrib.auth.models import User
+from rest_framework.exceptions import AuthenticationFailed
+
+class CustomAuthentication(BasicAuthentication):
+    def authenticate(self, request):
+        username = request.GET.get('username')
+        if username is None:
+            return None
```
Comment (Hardik0812) [inline rest_api/api/auth.py]: why return None ???? return a message that please provide username

File: rest_api/api/views.py [hunk 2]
```diff
@@ -278,17 +281,25 @@ def destroy(Self,request, pk):
...
-from rest_framework.permissions import IsAuthenticated, AllowAny , IsAdminUser
-
+from rest_framework.authentication import BasicAuthentication, SessionAuthentication
+from rest_framework.permissions import IsAuthenticated, AllowAny , IsAdminUser, IsAuthenticatedOrReadOnly,DjangoModelPermissions, DjangoModelPermissionsOrAnonReadOnly
+from api.permission import MyPermission
+from api.auth import CustomAuthentication
 
 class StudentModelViewSet(viewsets.ModelViewSet):
     queryset = Student.objects.all()
     serializer_class = StudentSerializer
```
Comment (Hardik0812) [inline rest_api/api/views.py:317]: remove all commented code from code

Comment (Kevalkukadiya) [reply rest_api/api/views.py:317]: Done

File: rest_api/api/auth.py (hunk 1 above)
Comment (Kevalkukadiya) [reply rest_api/api/auth.py]: Done

**Pull request merged**